src/katgpucbf/xbgpu/output.py

```diff
@@ -0,0 +1,45 @@
+################################################################################
+# Copyright (c) 2023, National Research Foundation (SARAO)
+#
+# Licensed under the BSD 3-Clause License (the "License"); you may not use
+# this file except in compliance with the License. You may obtain a copy
+# of the License at
+#
+#   https://opensource.org/licenses/BSD-3-Clause
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+"""Data structures capturing static configuration of a single output stream."""
+
+from abc import ABC
+from dataclasses import dataclass
+
+from katsdptelstate.endpoint import Endpoint
+
+
+@dataclass
+class Output(ABC):
+    """Static configuration for an output stream."""
+
+    name: str
+    dst: Endpoint
+    channels_per_substream: int
+
+
+@dataclass
+class XOutput(Output):
+    """Static configuration for an output X-engine stream."""
+
+    pass
```

Comment:
I'd suggest making the integration time (or whatever the internal representation of it is) a property of XOutput. In theory (although I can't really see a use case), you could have an engine produce two baseline-correlation-product streams with different integration times.

Reply:
As mentioned earlier, I'm absolutely for this. Fair warning, this comment did involve some ~overthinking on my part.

In going through implementing it, and with the context of [your review summation](https://github.com/ska-sa/katgpucbf/pull/568#pullrequestreview-1452387027), I guess I should conclude
* An Output class shall cart around input config items as long as they are of use and we can agree it's useful (?)

e.g. I did away with the `antennas` attribute on `XOutput` because the calculation of the `n_baselines` property really was for convenience. 

For the record, it depends on three input config items used for `timestamp_increment_per_accumulation`
* `heap_accumulation_threshold`
* `samples_between_spectra`, and
* `spectra_per_heap`.

That seems like a lot to me, but if it's needed then who am I to judge 🤷 I'll go ahead with 
```
class XOutput(Output):
    heap_accumulation_threshold: int
    samples_between_spectra: int
    spectra_per_heap: int

    @property
    def timestamp_increment_per_accumulation(self) -> int:
        .
        .
        .
```

Reply:
Not quite what I had in mind. After reviewing the current CLI, what I meant was to add just `heap_accumulation_threshold` to the class (that is the "integration time", just specified in a units of heaps).

Reply:
Oh dear, if only I'd seen this 30 seconds ago. Apologies, rolling back now.